components/gitpod-cli/cmd/rebuild.go

```diff
@@ -0,0 +1,232 @@
+	cmd := exec.Command("docker", "ps", "-q", "-f", "label=gp-rebuild")
+	containerIds, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+
+	for _, id := range strings.Split(string(containerIds), "\n") {
+		if len(id) == 0 {
+			continue
+		}
+
+		cmd = exec.Command("docker", "stop", id)
+		err := cmd.Run()
+		if err != nil {
+			return err
+		}
+
+		cmd = exec.Command("docker", "rm", "-f", id)
+		err = cmd.Run()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+var buildCmd = &cobra.Command{
+	Use:    "rebuild",
+	Short:  "Re-builds the workspace image (useful to debug a workspace custom image)",
+	Hidden: false,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+
+		client, err := supervisor.New(ctx)
+		if err != nil {
+			utils.LogError(ctx, err, "Could not get workspace info required to build", client)
+			return
+		}
+		defer client.Close()
+
+		wsInfo, err := client.Info.WorkspaceInfo(ctx, &api.WorkspaceInfoRequest{})
+		if err != nil {
+			utils.LogError(ctx, err, "Could not fetch the workspace info", client)
+			return
+		}
+
+		tmpDir, err := os.MkdirTemp("", "gp-rebuild-*")
+		if err != nil {
+			log.Fatal("Could not create temporary directory")
+			return
+		}
+		defer os.RemoveAll(tmpDir)
+
+		event := utils.TrackEvent(ctx, client, &utils.TrackCommandUsageParams{
+			Command: cmd.Name(),
+		})
+		defer event.Send(ctx)
+
+		gitpodConfig, err := utils.ParseGitpodConfig(wsInfo.CheckoutLocation)
+		if err != nil {
+			fmt.Println("The .gitpod.yml file cannot be parsed: please check the file and try again")
+			fmt.Println("")
+			fmt.Println("For help check out the reference page:")
+			fmt.Println("https://www.gitpod.io/docs/references/gitpod-yml#gitpodyml")
+			event.Set("ErrorCode", utils.RebuildErrorCode_MalformedGitpodYaml)
+			return
+		}
+
+		if gitpodConfig == nil {
+			fmt.Println("To test the image build, you need to configure your project with a .gitpod.yml file")
+			fmt.Println("")
+			fmt.Println("For a quick start, try running:\n$ gp init -i")
+			fmt.Println("")
+			fmt.Println("Alternatively, check out the following docs for getting started configuring your project")
+			fmt.Println("https://www.gitpod.io/docs/configure#configure-gitpod")
+			event.Set("ErrorCode", utils.RebuildErrorCode_MissingGitpodYaml)
+			return
+		}
+
+		var baseimage string
+		switch img := gitpodConfig.Image.(type) {
+		case nil:
+			baseimage = ""
+		case string:
+			baseimage = "FROM " + img
+		case map[interface{}]interface{}:
+			dockerfilePath := filepath.Join(wsInfo.CheckoutLocation, img["file"].(string))
+
+			if _, err := os.Stat(dockerfilePath); os.IsNotExist(err) {
+				fmt.Println("Your .gitpod.yml points to a Dockerfile that doesn't exist: " + dockerfilePath)
+				event.Set("ErrorCode", utils.RebuildErrorCode_DockerfileNotFound).Send(ctx)
+				return
+			}
+			dockerfile, err := os.ReadFile(dockerfilePath)
+			if err != nil {
+				log.Fatal("Could not read the Dockerfile")
```

Comment:
TODO: Add `event.Set("ErrorCode", ...)` here.

Reply:
Can this be caused by the user? I chose to logError to GCP instead of Segment because I imagined this issue being caused by a misconfiguration or something not working at workspace level rather than a mistake from the user

Reply:
I'm marking this comment as resolved, considering the answer at https://github.com/gitpod-io/gitpod/pull/15638/files#r1069205675.